### PR TITLE
rm: add support for removing empty dir with invalid permissions

### PR DESCRIPTION
**Pull request by shinhs0506**
add support for removing inaccessible/unreadable empty directories on unix platforms

I had to refactor `handle_dir` because it was becoming unreadable and it fixed other GNU tests where we previously had to alter its output.

~~I don't think the last 2 fails are regressions according to #2949, but can someone confirm?~~

also partially fixes `tests/rm/empty-inacc.sh`

### CODE REVIEW

**Comment by sylvestre** [comment]
both tests/misc/pwd-long & tests/rm/deep-2 were passing. Could you please have a look? thanks

File: src/uu/rm/src/rm.rs [hunk 1]
```diff
@@ -399,14 +408,54 @@ fn remove_dir(path: &Path, options: &Options) -> bool {
...
+
+    false
+}
+
+fn remove_empty_dir_no_perm(path: &Path, options: &Options) -> bool {
+    match fs::metadata(path) {
+        Ok(metadata) => {
+            let original_perm = metadata.permissions();
+            #[cfg(unix)]
+            fs::set_permissions(path, fs::Permissions::from_mode(0o777)).unwrap();
```
Comment (sylvestre) [inline src/uu/rm/src/rm.rs:432]: you aren't managing the potential error here, is it expected ?
thanks

Comment (shinhs0506) [reply src/uu/rm/src/rm.rs]: I should prob handle that

**Comment by shinhs0506** [comment]
resolves #3722

**Comment by sylvestre** [comment]
sorry we didn't merge this yet. I tried to rebase it, let's see if it works

**Comment by sylvestre** [comment]
We will need to fix the last three

**Comment by shinhs0506** [comment]
@sylvestre resolved, the lint error is fixed by #5360

**Comment by tertsdiepraam** [comment]
Oh I forgot about this PR while writing https://github.com/uutils/coreutils/pull/5344, we're definitely gonna have some difficult merge conflicts between those two. I'll probably rebase mine on top of yours, but I'll look at this PR some more first to see how much we can port over.

**Comment by sylvestre** [comment]
@tertsdiepraam you might need to update this one too :)

**Comment by sylvestre** [comment]
no activity for a while, closing

**Pull request closed without merging**